kubectl-pg/cmd/delete.go

```diff
@@ -0,0 +1,96 @@
+// Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1"
+	PostgresqlLister "github.com/zalando/postgres-operator/pkg/generated/clientset/versioned/typed/acid.zalan.do/v1"
+	"io/ioutil"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/scheme"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+)
+
+// deleteCmd represents kubectl pg delete.
+var deleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Deletes postgresql object by cluster-name/manifest file",
+	Long: `Deletes the postgres objects specific to a manifest file or cluster-name.
+Deleting the manifest is sufficient to delete the cluster.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		namespace, _ := cmd.Flags().GetString("namespace")
+		file, _ := cmd.Flags().GetString("file")
+		if file != "" {
+			deleteByFile(file)
+		} else if namespace != "" {
```

Comment:
if the `namespace` is not given, this command should delete a cluster in the current namespace.
`kubectl` is configurable for that with 
```bash
kubectl config set-context $(kubectl config current-context) --namespace="your-namespace"
```

Reply:
Yes this does exactly the same. If namespace is not provided we pick the current namespace the user is in. Else if a namespace is provided we try deleting the provided cluster in provided namespace.